sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1334,6 +1427,7 @@ pi_result _pi_context::getAvailableCommandList(
         (!ForcedCmdQueue ||
          *ForcedCmdQueue == CommandBatch.OpenCommandList->second.ZeQueue)) {
       CommandList = CommandBatch.OpenCommandList;
+      PI_CALL(Queue->insertStartBarrierIfDiscardEventsMode(CommandList));
```

Comment:
why there is no need to call for "insertActiveBarriers" on this path but need this barrier?

Reply:
We need to call insertStartBarrierIfDiscardEventsMode here because we need to insert this barrier whenever we start working with new command list (deferent from the last used) -  it doesn't matter if it was recycled or new command list was created or we just took existing open command list. In this branch we take existing open command list, it may be different from the last used command list, for example if there are from different engines.

According to my understanding insertActiveBarriers cares only about new command lists which may be potentiallu created in new L0 queues  where we haven't put  active barriers yet. If recycled or existing open command list is used then active barriers were already inserted on corresponding ze_queue before.

Reply:
> According to my understanding insertActiveBarriers cares only about new command lists which may be potentiallu created in new L0 queues where we haven't put active barriers yet. If recycled or existing open command list is used then active barriers were already inserted on corresponding ze_queue before.

Hmm, that's vague. We also have some known problems with "ActiveBarries" so maybe they are related.
I was going to suggest that we do "insertStartBarrierIfDiscardEventsMode" from "insertActiveBarriers".
But OK, let's keep it this way for now.